pdf/pdfdocument.cpp

```diff
@@ -315,9 +318,17 @@ void PDFDocument::jobFinished(PDFJob *job)
 
 void PDFDocument::searchFinished(const QList<QPair<int, QRectF>> &matches)
 {
-    delete d->searchModel;
-    d->searchModel = new PDFSearchModel(matches);
-    emit searchModelChanged();
+    if (matches.count() != d->searchModel->count())
```

Comment:
No, I've kept passing a reference to the complete list as a remnant of the previous way to do. So there is two way to catch the search results, either build it on the way with the progress signal, or at the end with the finished signal. The latter is not used anymore. If you prefer, I can remove it.

Additionally, we can also delete the list in the thread after the finished signal has been emitted, since the client have now kept a copy of it. I'm not sure this represent a big memory print though.